linode/manager
# Fix flaky Cypress volume tests, `cy.visitWithLogin(...)` improvements

## Description

**What does this PR do?**
* Adds the ability to override local storage, individual user preferences, and automatically mock common API requests via `cy.visitWithLogin()`, facilitating these improvements:
  * Volume tests now force the landing page to show up to 100 results, mitigating the issue we're having with stuck volumes and fixing all of the flakiness I'm currently aware of with regards to volumes
  * Some Linode and Volume smoke tests had an issue where the Akamai/Linode billing banner still appears even though it should be dismissed; allowing user preferences to be overridden individually resolves this issue, as mocks were causing the dismissed banner preference to be dropped
  * The `/account` API request is now mocked for all tests except some billing tests, dramatically improving test speeds for accounts where that endpoint is particularly slow

## How to test

**What are the steps to reproduce the issue or verify the changes?**
You can run Cypress to confirm that these changes do not break anything, but testing that they fix the issues as described is a little bit trickier if your test account doesn't already have a bunch of stuck volumes. Feel free to reach out to me on Slack, and I can help you set up your `.env` file with a test account that will enable proper testing.
